include/mitsuba/render/endpoint.h

```diff
@@ -115,6 +115,52 @@ class MTS_EXPORT_RENDER Endpoint : public Object {
                      const Point2f &sample,
                      Mask active = true) const;
 
+    /**
+     * \brief Independently of the scene, sample a position on the endpoint.
+     *
+     * This is entirely independent of transport phenomena, and samples
+     * uniformly over the surface area.
+     *
+     * For infinite area lights, the returned point is placed on a bounding
```

Comment:
Hrmmm, I must admit I am not a fan of this interface.

What I would prefer is the convention that envmap emitters only support the `sample_ray` interface that is actually more complex: it picks a direction and then an offset on a disk (which is the projection of the bounding sphere onto the sampled direction).

Reply:
That wold be nice if we can achieve it. For now this was used in `sample_visible_emitters` (emitters directly visible from the sensor). If we sample a ray from the emitter, there's zero probability it will connect with the sensor.

What I could do though is have a special case in the integrator: if it's an infinite emitter, only `sample_direction`, otherwise `sample_position`.

Reply:
Update: managed to get rid of `sample_position` for infinite emitters by adding a special case in the integrator.